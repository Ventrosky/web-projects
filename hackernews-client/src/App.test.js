import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App, { Search, Button, Table} from './App';


Enzyme.configure({ adapter: new Adapter() });

const emptyFu = () => { return null; };

describe('App', () => {

	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<App />, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

	test('has valid snapshot', () => {
		const component = renderer.create(
			<App />
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

});

describe('Search', () => {

	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<Search>Search</Search>, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

	test('has valid snapshot', () => {
		const component = renderer.create(
			<Search>Search</Search>
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

});

describe('Button', () => {

	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<Button onClick={(e) => e.preventDefault()}>Give Me More</Button>, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

	test('has valid snapshot', () => {
		const component = renderer.create(
			<Button onClick={emptyFu}>Give Me More</Button>
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should contain className', () => {
		const element = shallow(
			<Button 
                  onClick={emptyFu}
                  className="button-test"
            > 
                Test Button
            </Button>

		);
		expect(element.find('.button-test').length).toBe(1);
	});

});

describe('Table', () => {

	const props = {
		list: [
			{ title: '1', author: '1', num_comments: 1, points: 2, objectID: 'y' },
			{ title: '2', author: '2', num_comments: 1, points: 2, objectID: 'z' },
		],
		onDismiss: emptyFu,
	};

	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<Table { ...props } />, div);
	});

	test('has valid snapshot', () => {
		const component = renderer.create(
			<Table { ...props } />
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('shows two items in list', () => {
		const element = shallow(
			<Table { ...props } />
		);

		expect(element.find('.table-row').length).toBe(2);
	});

});
