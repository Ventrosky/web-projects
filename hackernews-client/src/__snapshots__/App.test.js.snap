// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has valid snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value="redux"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="table"
  >
    <div
      className="table-header"
    >
      <span
        style={
          Object {
            "width": "40%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
          type="button"
        >
          Title
          <span>
             
          </span>
          <i
            className=""
          />
        </button>
      </span>
      <span
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
          type="button"
        >
          Author
          <span>
             
          </span>
          <i
            className=""
          />
        </button>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
          type="button"
        >
          Comments
          <span>
             
          </span>
          <i
            className=""
          />
        </button>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
          type="button"
        >
          Points
          <span>
             
          </span>
          <i
            className=""
          />
        </button>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        Archive
      </span>
    </div>
  </div>
  <div
    className="interactions"
  >
    <div>
      <i
        className="fa fa-spinner fa-spin"
      />
    </div>
  </div>
</div>
`;

exports[`Button has valid snapshot 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  Give Me More
</button>
`;

exports[`Search has valid snapshot 1`] = `
<form
  onSubmit={undefined}
>
  <input
    onChange={undefined}
    type="text"
    value={undefined}
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;

exports[`Table has valid snapshot 1`] = `
<div
  className="table"
>
  <div
    className="table-header"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Title
        <span>
           
        </span>
        <i
          className=""
        />
      </button>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Author
        <span>
           
        </span>
        <i
          className=""
        />
      </button>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Comments
        <span>
           
        </span>
        <i
          className=""
        />
      </button>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Points
        <span>
           
        </span>
        <i
          className=""
        />
      </button>
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      Archive
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href={undefined}
      >
        1
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
       
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href={undefined}
      >
        2
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
       
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      1
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
       
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <button
        className="button-inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
</div>
`;
